derive/src/trait_bounds.rs

```diff
@@ -39,12 +38,19 @@ use crate::{
 /// relevant generic types including associated types (e.g. `T::A: TypeInfo`), correctly dealing
 /// with self-referential types.
 ///
-/// Ignores any type parameters not included in `type_params`.
+/// # Effect of attributes
+///
+/// `#[scale_info(skip_type_params(..))]`
+///
+/// Will not add `TypeInfo` bounds for any type parameters skipped via this attribute.
+///
+/// `#[scale_info(bounds(..))]`
+///
+/// Replaces *all* auto-generated trait bounds with the user-defined ones.
```

Comment:
```rust
#[scale_info(skip_type_params(U))]
#[scale_info(bounds("U: Other + Traits + NotTypeInfoStuff))]
struct A<T, U> { … }
```

In the above, I will not get `TypeInfo` bounds auto-generated for `T` (because `bounds()` override **all** auto generating machinery) but I will get the bounds I manually specified for `U` (because it's `bounds(…)` trumps `skipe_type_params`). Is that correct?

A few examples like above would be good! :)

Reply:
> In the above, I will not get TypeInfo bounds auto-generated for T (because bounds() override all auto generating machinery) but I will get the bounds I manually specified for U (because bounds(…) trumps skip_type_params). Is that correct?

Correct. It trumps it in respect to the generated `where` clause. However `skip_type_params` still has its effect on the generated list of `type_params` for the type.

> A few examples like above would be good! :)

I am in the process of adding more rustdocs on the topic, in the meantime I have added some more ui tests and have updated the PR description.